### PR TITLE
cpu: aarch64: matmul: Move allocation of temporary tensors to scratchpad in acl_matmul

### PR DESCRIPTION
**Pull request by annop-w**
# Description

Introduce 3 new scrathpad memory key names.

# Checklist

## General

- [x] Do all unit and benchdnn tests (`make test` and `make test_benchdnn_*`) pass locally for each commit?
- [x] Have you formatted the code using clang-format?

### CODE REVIEW

File: src/cpu/aarch64/matmul/acl_matmul_utils.hpp [hunk 1]
```diff
@@ -63,7 +63,8 @@ status_t init_conf_matmul(acl_matmul_conf_t &amp, memory_desc_t &src_md,
         const primitive_attr_t &attr);
 
 status_t init_scratchpad(memory_tracking::registrar_t &scratchpad,
-        acl_matmul_conf_t &amp, memory_desc_t &dst_md);
+        acl_matmul_conf_t &amp, memory_desc_t &src_md,
+        memory_desc_t &weights_md, memory_desc_t &dst_md);
```
Comment (dzarukin) [inline src/cpu/aarch64/matmul/acl_matmul_utils.hpp:67]: `const` for every input object?

**Review by dzarukin — APPROVED** [review]

File: src/cpu/aarch64/matmul/acl_matmul_utils.hpp (hunk 1 above)
Comment (annop-w) [reply src/cpu/aarch64/matmul/acl_matmul_utils.hpp]: Done.

**Comment by mgouicem** [comment]
tagging @snadampal as this relates to #1470

**Comment by vpirogov** [comment]
+@snadampal, could you please help reviewing these changes?

@annop-w, please resolve merge conflict.

**Comment by annop-w** [comment]
> @annop-w, please resolve merge conflict.

Done.

File: src/common/memory_tracking.hpp [hunk 2]
```diff
@@ -1,5 +1,6 @@
 /*******************************************************************************
 * Copyright 2018-2024 Intel Corporation
+* Copyright 2024 Arm Ltd. and affiliates
```
Comment (dzarukin) [inline src/common/memory_tracking.hpp:3]: Seems to be unneeded.

**Comment by snadampal** [comment]
glad to see this change finally coming.
Hi @annop-w  to understand the change a bit in detail, the scratchpad buffers for src and weights are the same buffers allocated in the framework, right . for example, by ideep in PyTorch and by mkldnn wrapper in TensorFlow?

thanks @mgouicem and @vpirogov for tagging me here.

File: src/common/memory_tracking.hpp (hunk 2 above)
Comment (annop-w) [reply src/common/memory_tracking.hpp:3]: I believe it should have been added in this previous PR https://github.com/oneapi-src/oneDNN/pull/1966

**Comment by annop-w** [comment]
@snadampal I am not sure how the scratchpads are currently managed in ideep (or Tensorflow), but the idea here is to allow for users (i.e. PyTorch or TF) a chance to decide that, isn't it ? For example, PyTorch can now choose to allocate the same buffer for both src and wei, if sensible, which was not possible before. Does this help ?

**Comment by snadampal** [comment]
i mean reusing the same buffer allocated in PT or TF via oneDNN user mode scratchpad, you clarified it, thanks. 
this change LGTM.

**Comment by annop-w** [comment]
@snadampal Ah, yes, in that case, you're absolutely right. Thanks for the review.

**Comment by vpirogov** [comment]
Awesome. Thanks, @snadampal.

**Pull request merged**